packages/blank-theme/src/lib/strapi.ts

```diff
@@ -0,0 +1,326 @@
+            ...(revalidate
+               ? {
+                    next: {
+                       revalidate: revalidate
+                    }
+                 }
+               : {})
+         }
+      })
+
+      if (!response.ok) {
+         throw new Error(`Failed to fetch data: ${response.statusText}`)
+      }
+
+      const data = await response.json()
+      return { data, error: null }
+   } catch (error: any) {
+      console.error(`Error during API call: ${error.message}`)
+      return {
+         data: null,
+         error: error.message || "An error occurred during data fetch"
+      }
+   }
+}
+
+/**
+ * Function to get data
+ * @param model  model name as like strapi model name `pages`, `messages`...
+ * @param query  query object to filter data
+ * @param cache  cache type `force-cache`, `no-cache`, `no-store`
+ * @param revalidate  revalidate time in seconds
+ * @returns  {data, error}
+ * @example
+ * const { data, error } = await find("pages", { filters: { slug: { $eq: "home" } } }, "force-cache", 60);
+ */
+export const find = async (
+   model: string,
+   query: any = {},
+   cache: "force-cache" | "no-cache" | "no-store" = "force-cache",
+   revalidate?: number
+) => {
+   const queryString = qs.stringify(query, {
+      arrayFormat: "indices",
+      encode: false,
+      indices: false
+   })
+
+   try {
+      const response = await fetch(`${apiUrl}/${model}/?${queryString}`, {
+         method: "GET",
+         headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${process.env.STRAPI_AUTH_TOKEN}`
+         },
+         ...{
+            ...(revalidate ? {} : { cache }),
+            ...(revalidate
+               ? {
+                    next: {
+                       revalidate: revalidate
+                    }
+                 }
+               : {})
+         }
+      })
+
+      if (!response.ok) {
+         throw new Error(`Failed to fetch data: ${response.statusText}`)
+      }
+
+      const data = await response.json()
+
+      return { data, error: null }
+   } catch (error: any) {
+      console.error(`Error during API call: ${error.message}`)
+      return {
+         data: null,
+         error: error.message || "An error occurred during data fetch"
+      }
+   }
+}
+
+/**
+ * Function to delete an entry by ID
+ * @param model  model name as in Strapi model name, e.g., `restaurants`, `pages`, `messages`
+ * @param id     ID of the entry to delete
+ * @returns  {data, error} Returns the deleted entry data or an error if the operation fails
+ * @example
+ * const { data, error } = await deleteEntry("restaurants", "1");
+ */
+
+export const deleteEntry = async (model: string, id: number) => {
+   try {
+      const response = await fetch(`${apiUrl}/${model}/${id}`, {
+         method: "DELETE",
+         headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${process.env.STRAPI_AUTH_TOKEN}` // Replace with your token
```

Comment:
we are not creating any strapi token that delete entry. I belive it will be user token `STRAPI_AUTH_TOKEN}`